### PR TITLE
Feat: v3 support

### PR DESCRIPTION
**Pull request by Meldiron**
Adds support for communication with both `v2` and `v3` of Open Runtime.

### CODE REVIEW

File: tests/resources/functions/ruby/index.rb [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
+    context.log('Sample Log')
     
-    return res.json({
+    return context.res.json({
         'isTest': true,
         'message': 'Hello Open Runtimes 👋',
         'todo': todo,
-        'variable': req.variables['test-variable']
+        'url': context.req.url,
+        'variable': ENV['TEST_VARIABLE'] || nil,
```
Comment (christyjacob4) [inline tests/resources/functions/ruby/index.rb:14]: what's `ENV`?  where is it set?

File: app/http.php [hunk 2]
```diff
@@ -342,11 +344,20 @@ function removeAllRuntimes(Table $activeRuntimes, Pool $orchestrationPool): void
...
+                    'OPEN_RUNTIMES_ENTRYPOINT' => $entrypoint,
+                    'OPEN_RUNTIMES_HOSTNAME' => System::getHostname()
+                ]);
+            } else {
+                $variables = \array_merge($variables, [
+                    'INTERNAL_RUNTIME_KEY' => $secret,
+                    'INTERNAL_RUNTIME_ENTRYPOINT' => $entrypoint,
+                    'INERNAL_EXECUTOR_HOSTNAME' => System::getHostname()
+                ]);
+            }
```
Comment (christyjacob4) [inline app/http.php:359]: You can use the better `match` syntax to clean up this code :)

File: app/http.php [hunk 3]
```diff
@@ -278,10 +279,11 @@ function removeAllRuntimes(Table $activeRuntimes, Pool $orchestrationPool): void
     ->param('remove', false, new Boolean(), 'Remove a runtime after execution.', true)
     ->param('cpus', 1, new Integer(), 'Container CPU.', true)
     ->param('memory', 512, new Integer(), 'Comtainer RAM memory.', true)
+    ->param('version', 'v3', new WhiteList(['v2', 'v3']), 'Runtime Open Runtime version.', true)
```
Comment (christyjacob4) [inline app/http.php:282]: May be a good idea to declare these as constants somewhere

Comment (christyjacob4) [reply app/http.php:282]: what's your opinion on using a different endpoint for v3 runtimes ? currently the endpoint is `v1/runtimes`. we could also use `/v2/runtimes` .

File: app/http.php [hunk 4]
```diff
@@ -539,19 +549,23 @@ function removeAllRuntimes(Table $activeRuntimes, Pool $orchestrationPool): void
     ->desc('Create an execution')
     // Execution-related
     ->param('runtimeId', '', new Text(64), 'The runtimeID to execute.')
-    ->param('payload', '', new Text(8192), 'Data to be forwarded to the function, this is user specified.', true)
-    ->param('variables', [], new Assoc(), 'Environment variables passed into runtime.', true)
+    ->param('body', '', new Text(20971520), 'Data to be forwarded to the function, this is user specified.', true)
```
Comment (christyjacob4) [inline app/http.php:552]: on what basis is this limit set? what if the payload exceeds this size? For example when uploading a file in chunks ? Can we make sure it accepts 5MB ?

File: tests/resources/functions/cpp/index.cc [hunk 5]
```diff
@@ -1,66 +0,0 @@
-#include <iostream>
-#include <string>
-#include <curl/curl.h>
-
-static size_t WriteCallback(void *contents, size_t size, size_t nmemb, void *userp)
```
Comment (christyjacob4) [inline tests/resources/functions/cpp/index.cc:5]: any reason we deleted this file ?

File: tests/ExecutorTest.php [hunk 6]
```diff
@@ -190,40 +188,47 @@ public function provideScenarios(): array
...
+                    $this->assertEquals(200, $response['headers']['status-code']);
+                    $this->assertEquals(200, $response['body']['statusCode']);
+                    $this->assertEquals('{"message":"Hello Open Runtimes 👋"}', $response['body']['body']);
+                    $this->assertEmpty($response['body']['logs']);
+                    $this->assertEmpty($response['body']['errors']);
                 }
             ],
             [
-                'image' => 'openruntimes/node:v2-18.0',
+                'image' => 'openruntimes/node:v3-18.0',
```
Comment (christyjacob4) [inline tests/ExecutorTest.php:202]: Now that we have both v3 and v2 images, does the warm up task in the beginning also pull the v3 images?

**Review by christyjacob4 — CHANGES_REQUESTED** [review]
Looks good. Please address the comments

File: app/http.php (hunk 3 above)
Comment (Meldiron) [reply app/http.php:282]: I feel like that would introduce a lot of code duplication. Most of the code stays the same, just the way request is built is different (and how response is parsed).

But if you see a benefit in it, we can surely do it.

File: app/http.php (hunk 4 above)
Comment (Meldiron) [reply app/http.php:552]: This should support up to 20MB, which is limit that we consider reasonable even in Appwrite.
5MB chunk uploads should work just fine. Trying to upload 30MB file at once will fail.

File: app/http.php [hunk 7]
```diff
@@ -699,66 +722,180 @@ function (string $runtimeId, string $payload, array $variables, int $timeout, st
...
+                        'errors' => '',
+                        'headers' => []
+                    ];
+                }
+
+                // Extract response
+                $executorResponse = json_decode(\strval($executorResponse), false);
+
+                $res = '';
+                if ($statusCode >= 200 && $statusCode < 500) {
```
[the remark this answers is not in the document]
Comment (Meldiron) [reply app/http.php:741]: Agree. Updated.

File: tests/ExecutorTest.php (hunk 6 above)
Comment (Meldiron) [reply tests/ExecutorTest.php:202]: Only v3 images. v2 are no longer pulled.

File: tests/resources/functions/cpp/index.cc (hunk 5 above)
Comment (Meldiron) [reply tests/resources/functions/cpp/index.cc:5]: C++ wasnt finished when I started this branch.

Re-added CPP tests.

**Review by christyjacob4 — APPROVED** [review]

**Pull request merged**